Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderBuildArgumentsManager.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ShaderBuildArgumentsManager.h"
+
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+
+#include <Atom/RPI.Edit/Common/JsonUtils.h>
+#include <Atom/RHI.Edit/ShaderBuildOptions.h>
+
+namespace AZ
+{
+    namespace ShaderBuilder
+    {
+        // FIXME: 
+        static AZStd::string ResolvePathMacros(AZStd::string_view inPath)
+        {
+            AZStd::string resolvedPath(inPath);
+            AZ::StringFunc::Replace(resolvedPath, "@PROJECTROOT@", AZ::Utils::GetProjectPath().c_str());
+            AZ::StringFunc::Replace(resolvedPath, "@PROJECTNAME@", AZ::Utils::GetProjectName().c_str());
+            AZ::StringFunc::Replace(resolvedPath, "@ENGINEROOT@", AZ::Utils::GetEnginePath().c_str());
+            AZ::StringFunc::Path::Normalize(resolvedPath);
+            return resolvedPath;
+        }
+
+        void ShaderBuildArgumentsManager::Init()
+        {
+            AZStd::unordered_map<AZStd::string, AZ::RHI::ShaderBuildArguments> removeBuildArgumentsMap;
+            AZStd::unordered_map<AZStd::string, AZ::RHI::ShaderBuildArguments> addBuildArgumentsMap;
+            auto configFiles = DiscoverConfigurationFiles();
```

Comment:
Do we have an issue currently where this initialization won't rerun if the config file changes?

Reply:
ShaderBuildArgumentsManager::Init() always runs when ProcessJob() starts.